odl/phantom/noise.py

```diff
@@ -225,7 +225,7 @@ def salt_pepper_noise(vector, fraction=0.05, salt_vs_pepper=0.5,
                       for subintensity in vector]
         else:
             # Extract vector of values
-            values = as_flat_array(vector).copy()
+            values = vector.asarray().flatten()
```

Comment:
Flatten is really slow and always copies, do we want that?

Reply:
Here we explicitly want a copy at the beginning.